### ISSUE TITLE
Composer install tries and fails to install pre-commit hook (GIT submodule)

### ISSUE THREAD

**Issue by BrixCRM** [mautic/mautic#3372]
What type of report is this:

| Q  | A
| ---| ---
| Bug report? | Y
| Feature request? | 
| Enhancement? | Y

## Description:
We are used to GIT submodules, and I have added Mautic to my GIT project as a submodule.
Now the .git is not a folder, but a 'gitlink'; iow. a file containing ``gitdir: ../.git/modules/mautic``

This breaks the scripts in ``composer.json``. To script this, it needs to ask GIT about the .git folder location:

```
git rev-parse --git-dir
```

## If a bug:

| Q   | A
| --- | ---
| Mautic version | 2.5.1 / master
| PHP version | 5.6.x

### Steps to reproduce:
1. add mautic as submodule: ``git submodule add git@github.com:mautic/mautic.git``
2. run ``composer install`` in submodule prefix folder
 
### Log errors: 

```
> cp ./build/hooks/pre-commit ./.git/hooks/pre-commit
cp: ./.git/hooks/pre-commit: Not a directory
Script cp ./build/hooks/pre-commit ./.git/hooks/pre-commit handling the post-install-cmd event returned with error code 1
```

**Comment by BrixCRM** [comment]
I can easily prepare a PR but:

- what branch do I commit it on, I am not sure
- testing; I can run the script by hand. But I have just a fresh, std. checkout with no changes, so have to assume its running ok.

**Comment by MarkLL** [comment]
actually if you check https://github.com/mautic/mautic/pull/3157 I've already proposed a fix that works in windows as well. Please try it and comment. Your proposed fix will not work on all platforms!

**Comment by BrixCRM** [comment]
Actually I've seen #3157, but it does not mention GIT submodules. 

- There is no issue for me with .git/hooks/pre-commit script, but with the scripts in composer.json. #3157 seemed to have to do with the pre-commit script lines itself. Not the path.
- GIT rev-parse --git-dir is supported since quite some versions; see https://git-scm.com/docs/git-rev-parse. I am not aware of a reason why windows GIT client would not support this.
- I have looked at the changeset, but it seems to solve the issue by duplicating the GIT hook script. But I cannot see what its exact changeset it has vs. the original script.

In short, I cannot find the actual cause of your error in the report. What exactly does not work? The composer script, or the pre-commit? do you know why?

I would personally solve this by using Sh-compatible scripts. But does windows composer run in some Powershell?

**Comment by MarkLL** [comment]
@BrixCRM there are 2 issues that my mods fix. 1 is that the `cp` command is nix only. it just does not work in windows. 2. this method errors when the .git path is missing (download the zip and run composer install) 

The additional script that I added is because the git command must point to a .bat file for it to work on windows.

**Comment by BrixCRM** [comment]
@MarkLL its been a while since I used windows or MingW, I thought I saw a variation on the Sh script. 

First now I need to see about some testing/CI setup for our dev line, I'll keep these issues/PR's in mind then.
